src/Microsoft.DocAsCode.MarkdigEngine.Extensions/TripleColon/ImageExtensionInline.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+namespace Microsoft.DocAsCode.MarkdigEngine.Extensions
+{
+    using Markdig.Renderers;
+    using Markdig.Renderers.Html;
+    using Markdig.Syntax;
+    using Markdig.Syntax.Inlines;
+    using System;
+    using System.Collections.Generic;
+
+    public class ImageExtensionInline : ITripleColonExtensionInlineInfo
```

Comment:
A lot of code in here looks duplicated with `ImageExtensionBlock`, I wonder if it is possible to have a structure like:

```csharp
interface ITripleColon
{
    IDictionary<string, string> RenderProperties { get; }
    ITripleColonExtensionInfo Extension { get; }
    // other properties currently in TripleColonBlock
}

class TripleColonBlock : ContainerBlock, ITripleColon { }
class TripleColonInline : ContainerInline, ITripleColon { }

interface ITripleColonExtensionInfo
{
    bool SupportsInline { get; }
    bool Render(HtmlRenderer renderer, ITripleColon obj);
    // other properties in ITripleColonExtensionInfo
}

class ImageExtension : ITripleColonExtensionInfo { }
class CodeExtension : ITripleColonExtensionInfo { }
// ...
```